### PR TITLE
Enable semi mask

### LINKED ISSUE #3541: Optimization: Re-enable semi mask
### Description

Semi mask is disabled in #3524 to simplify the rework. See https://github.com/kuzudb/kuzu/pull/3524#discussion_r1612126572.

### PR DESCRIPTION
**Pull request by andyfengHKU**
# Description

This PR re-enables semi mask after our refactor on node table scan.

Fixes #3541 

# Contributor agreement

- [x] I have read and agree to the [Contributor Agreement](https://github.com/kuzudb/kuzu/blob/master/CLA.md).

### CODE REVIEW

File: src/include/storage/store/table.h [hunk 1]
```diff
@@ -22,6 +22,7 @@ struct TableScanState {
     std::unique_ptr<TableDataScanState> dataScanState;
     common::node_group_idx_t nodeGroupIdx = common::INVALID_NODE_GROUP_IDX;
 
+    // Zone map
```
Comment (ray6080) [inline src/include/storage/store/table.h:25]: Either remove the comment or be more explicit to explain the usage of `columnPredicateSets` and `zoneMapResult`.

File: src/include/storage/store/node_table.h [hunk 2]
```diff
@@ -22,14 +23,16 @@ class LocalNodeTable;
 struct NodeTableScanState final : TableScanState {
     // Local storage node group.
     LocalNodeNG* localNodeGroup = nullptr;
+    // Semi mask
```
Comment (ray6080) [inline src/include/storage/store/node_table.h:26]: Remove the comment here.

**Review by ray6080 — APPROVED** [review]
Can u also add into the PR description that now node scan applies vector-level semi mask instead of node-level? Also, we should run some benchmarks over the comparison of node-level vs. vector-level. maybe we can grab some reported numbers from our benchmark server

**Comment by andyfengHKU** [comment]
> Can u also add into the PR description that now node scan applies vector-level semi mask instead of node-level? Also, we should run some benchmarks over the comparison of node-level vs. vector-level. maybe we can grab some reported numbers from our benchmark server

I was going to do some benchmark. But then I realize our benchmark report already shows improvement.

**Pull request merged**